cmd/tape/main.go

```diff
@@ -50,3 +54,24 @@ func main() {
 	}
 	os.Exit(0)
 }
+
+func argsCheck(rate *float64, burst *int, logger *log.Logger) {
+	if *rate < 0 {
+		os.Stderr.WriteString("tape: error: rate must be zero as unlimited or positive\n")
+		os.Exit(1)
+	} else {
+		if *rate != 0 {
+			logger.Warnf("Will use rate %f as send rate\n", *rate)
+		}
+		if int64(*rate) > int64(*burst) {
+			fmt.Printf("As rate %d is bigger than burst %d, real rate is burst\n", int64(*rate), int64(*burst))
+		}
+	}
+	if *burst < 1 {
+		os.Stderr.WriteString("tape: error: burst at least 1\n")
+		os.Exit(1)
+	}
+	if *burst != 1000 {
+		logger.Warnf("Will use %d as burst\n", burst)
+	}
+}
```

Comment:
this could roughly looks like:
```go
func argsCheck(rate float64, burst int) {
	if rate < 0 {
		os.Stderr.WriteString("Rate must be zero (unlimited) or positive number\n")
		os.Exit(1)
	}

	if burst < 1 {
		os.Stderr.WriteString(fmt.Sprintf("Burst [%d] must be greater or equal to 1\n", burst))
		os.Exit(1)
	}

	if int64(rate) > int64(burst) {
		fmt.Printf("Rate %d is greater than burst %d, actual rate will be burst", int64(rate), int64(burst))
	}

	os.Stdout.WriteString(fmt.Sprintf("Send traffic at rate %d with burst %d", rate, burst))
}
```

Reply:
try to write less nested code to improve readability

Reply:
right